ambry-clustermap/src/main/java/com/github/ambry/clustermap/RecoveryTestClusterManager.java

```diff
@@ -52,7 +53,19 @@ public class RecoveryTestClusterManager implements ClusterMap {
   public PartitionId getPartitionIdFromStream(InputStream stream) throws IOException {
     DuplicatingInputStream duplicatingInputStream = new DuplicatingInputStream(stream);
     duplicatingInputStream.mark(0);
-    PartitionId partitionIdStatic = staticClusterManager.getPartitionIdFromStream(duplicatingInputStream);
+    PartitionId partitionIdStatic = null;
+    try {
+      partitionIdStatic = staticClusterManager.getPartitionIdFromStream(duplicatingInputStream);
+    } catch (IOException e) {
+      // for composite blob, it has the list of Blob IDs. It may not exist in the static cluster map.
```

Comment:
Please log an exception here or move the trace msg that says trying here before actually trying the helixMap

Reply:
I think log when partitionIdStatis is null is more generic. It can cover two cases,
1. staticClusterManager.getPartitionIdFromStream throws exception.
2. staticClusterManager.getPartitionIdFromStream returns null for some reason.

Reply:
But you are right, without the log here, we may not know exception ever happened. I'll add the extra log here.

Reply:
cool thanks